src/html2pdf.js

```diff
@@ -32,356 +32,412 @@
+
+            // Create and attach the elements.
+            var overlay = createElement('div', {className: 'html2pdf__overlay', style: overlayCSS});
+            var container = createElement('div', {className: 'html2pdf__container', style: containerCSS});
+            container.appendChild(source);
+            overlay.appendChild(container);
+            document.body.appendChild(overlay);
+
+            // Enable page-breaks.
+            var pageBreaks = source.querySelectorAll('.html2pdf__page-break');
+            var pxPageHeight = pageSize.inner.height * pageSize.k / 72 * 96;
+            Array.prototype.forEach.call(pageBreaks, function (el) {
+                el.style.display = 'block';
+                var clientRect = el.getBoundingClientRect();
+                el.style.height = pxPageHeight - (clientRect.top % pxPageHeight) + 'px';
+            }, this);
+
+            // Return the container.
+            return container;
+        };
+
+        html2pdf.makePDF = function (canvas, pageSize, opt) {
+            // Calculate the number of pages.
+            var ctx = canvas.getContext('2d');
+            var pxFullHeight = canvas.height;
+            var pxPageHeight = Math.floor(canvas.width * pageSize.inner.ratio);
+            var nPages = Math.ceil(pxFullHeight / pxPageHeight);
+            var pageModulus = pxFullHeight % pxPageHeight;
+
+            // Create a one-page canvas to split up the full image.
+            var pageCanvas = document.createElement('canvas');
+            var pageCtx = pageCanvas.getContext('2d');
+            var pageHeight = pageSize.inner.height;
+            pageCanvas.width = canvas.width;
+            pageCanvas.height = pxPageHeight;
+
+            // Initialize the PDF.
+            var pdf = new jsPDF(opt.jsPDF);
+
+            /**
+             * added support for page split only on white "rows".
+             * check if pixel is not "white" if it is less than whiteColorLevel value for the rgb values
+             * if the row isn't "white" - will go one pixel up and test that row, up to max. number of maxPixelToGoUp rows up
+             *
+             */
+            var whiteColorLevel = 150;
+            var maxPixelToGoUp = 150;
+            var pageStartPos = 0;
+            for (var page = 0; page < nPages; page++) {
+                // Trim the final page to reduce file size.
+                if (page === nPages - 1) {
+                    pageCanvas.height = pageModulus;
+                    pageHeight = pageCanvas.height * pageSize.inner.width / pageCanvas.width;
+                }
+                // Display the page.
+                var w = pageCanvas.width;
+                var heightDrawBack = 0;
+                do {
+
+                    var h = pageCanvas.height - heightDrawBack;
+                    var whiteRow = true;
+                    var wPix = 0;
+                    var bPixCount = 0;
+
+                    while (wPix < w && whiteRow) {
```

Comment:
This scan could be optimised, e.g. using `getImageData` on the entire row.

Reply:
i didn't have the time to optimize it - so if you have an idea of how to optimise it - please do!

Reply:
i've commited optimized solution for the white row checking